realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -303,4 +334,5 @@ static void simulateClientReset(SyncSession session) {
     protected static native void nativeInitializeSyncManager(String syncBaseDir);
     private static native void nativeReset();
     private static native void nativeSimulateSyncError(String realmPath, int errorCode, String errorMessage, boolean isFatal);
+    private static native void reconnect();
```

Comment:
I was mostly worried that the NetworkListener might not catch all cases that would be considered "offline", but for a public `reconnect()` to be truly useful we would need to expose other methods like `isOnline()` and `getReconnectDelay()` (or similar).

But I'm fine with just keeping it private for now, then we can evaluate from there.